stmoab/SummaryDashboard.py

```diff
@@ -170,20 +212,25 @@ def _add_query_to_dashboard(self, query_title, query_string,
                               column_mapping=None, series_options=None,
                               time_interval=None, stacking=True):
 
-    query_id, table_id = self.redash.create_new_query(
-        query_title, query_string, data_source)
-    viz_id = self.redash.create_new_visualization(
-        query_id,
-        visualization_type,
-        visualization_name,
-        chart_type,
-        column_mapping,
-        series_options,
-        time_interval,
-        stacking,
-    )
-    self.redash.add_visualization_to_dashboard(
-        self._dash_id, viz_id, visualization_width)
+    try:
+      query_id, table_id = self.redash.create_new_query(
+          query_title, query_string, data_source)
+      viz_id = self.redash.create_new_visualization(
+          query_id,
+          visualization_type,
+          visualization_name,
+          chart_type,
+          column_mapping,
+          series_options,
+          time_interval,
+          stacking,
+      )
+      self.redash.add_visualization_to_dashboard(
+          self._dash_id, viz_id, visualization_width)
+    except self.redash.RedashClientException as e:
+      raise self.ExternalAPIError(
+        "Unable to add query named {title} to "
```

Comment:
I would actually split this into three try/excepts with separate messages for each of the steps:

Unable to create query: {error}

Unable to create visualization: {error}

Unable to add visualization to dashboard: {error}